"ionic capacitor add android" fails due to unsatisfied peer dependency

**Description:**
<!-- Please describe the problem you're having. -->
Trying to run `ionic capacitor add android` on a project created a few months ago fails with `unable to resolve dependency tree` error.
The problem is I'm using `@capacitor/core@3.3.4` and `@capacitor/android@3.3.4` (already installed in node_modules folder).  

`@capacitor/android@3.3.4` has a peer dependency on `@capacitor/core@3.3.4` . But when I run `ionic capacitor add android`, it tries to install `@latest` instead of using the locally installed package, which results in `Could not resolve dependency: peer @capacitor/core@"^3.4.0" from @capacitor/android@3.4.3` since I have `@capacitor/core@3.3.4` installed.

Using capacitorjs CLI works just fine (`npx cap add android`)

Is this a bug or is it intendend behaviour? I found it weird I'm not able to keep using the CLI unless I update to the latest version, but maybe I'm missing something. 

**Steps to Reproduce:**
1. Install an old version of @capacitor/core and @capacitor/android:
```json
// package.json
{
    ...
    "dependencies": {
        "@capacitor/android": "3.3.4",
        "@capacitor/core": "3.3.4",
    },
   ...
}
```
2. Run `ionic capacitor add android`


**Output:**
```
$ npx ionic capacitor add android

> npm.cmd i -E @capacitor/android@latest
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: <appname>@0.0.1
npm ERR! Found: @capacitor/core@3.3.4
npm ERR! node_modules/@capacitor/core
npm ERR!   @capacitor/core@"3.3.4" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @capacitor/core@"^3.4.0" from @capacitor/android@3.4.3
npm ERR! node_modules/@capacitor/android
npm ERR!   @capacitor/android@"3.4.3" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See C:\Users\...\AppData\Local\npm-cache\eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\...\AppData\Local\npm-cache\_logs\2022-03-08T23_54_04_603Z-debug-0.log
[ERROR] An error occurred while running subprocess npm.

        npm.cmd i -E @capacitor/android@latest exited with exit code 1.

        Re-running this command with the --verbose flag may provide more information.
```

**My `ionic info`:**
```
[WARN] Error loading @capacitor/ios package.json: Error: Cannot find module '@capacitor/ios/package'

       Require stack:
       - C:\Users\lucas\AppData\Roaming\nvm\v16.14.0\node_modules\@ionic\cli\lib\project\index.js
       - C:\Users\lucas\AppData\Roaming\nvm\v16.14.0\node_modules\@ionic\cli\lib\index.js
       - C:\Users\lucas\AppData\Roaming\nvm\v16.14.0\node_modules\@ionic\cli\index.js
       - C:\Users\lucas\AppData\Roaming\nvm\v16.14.0\node_modules\@ionic\cli\bin\ionic

Ionic:

   Ionic CLI                     : 6.18.2 (C:\Users\lucas\AppData\Roaming\nvm\v16.14.0\node_modules\@ionic\cli)
   Ionic Framework               : @ionic/angular 6.0.2
   @angular-devkit/build-angular : 13.0.4
   @angular-devkit/schematics    : 13.0.4
   @angular/cli                  : 13.0.4
   @ionic/angular-toolkit        : 5.0.3

Capacitor:

   Capacitor CLI      : 3.3.4
   @capacitor/android : 3.3.4
   @capacitor/core    : 3.3.4
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 1.5.0

System:

   NodeJS : v16.14.0 (C:\Program Files\nodejs\node.exe)
   npm    : 8.3.1
   OS     : Windows 10
```


**Other Information:**
<!--
e.g. stacktraces, related issues, suggestions how to fix, stackoverflow links,
forum links, etc
-->

--- a/packages/@ionic/cli/src/commands/capacitor/base.ts
+++ b/packages/@ionic/cli/src/commands/capacitor/base.ts
@@ -273,7 +273,7 @@ export abstract class CapacitorCommand extends Command {
     }
 
     if (semver.gte(version, '3.0.0-alpha.1')) {
-      const [ manager, ...managerArgs ] = await pkgManagerArgs(this.env.config.get('npmClient'), { command: 'install', pkg: `@capacitor/${platform}@latest`, saveDev: false });
+      const [ manager, ...managerArgs ] = await pkgManagerArgs(this.env.config.get('npmClient'), { command: 'install', pkg: `@capacitor/${platform}@${version}`, saveDev: false });
       await this.env.shell.run(manager, managerArgs, { cwd: this.integration.root });
     }
 

